stdlib/date.rb

```diff
@@ -596,6 +598,40 @@ def tuesday?
     wday == 2
   end
 
+  def step(limit, step = 1, &block)
+    steps_count = (limit - self).to_i
+
+    if (steps_count * step) < 0
+      steps = []
+    else
+      if steps_count < 0
+        puts "steps_count -: #{steps_count}"
+        steps = (0 .. -steps_count).step(step.abs).map {|i| -i}.reverse
```

Comment:
Please,
1. Remove space from the range definition (i.e. `(0..-steps_count)`
2. Add spaces around curly braces (i.e. `.map { |i| -i }`